src/microbe_stage/biomes.h

```diff
@@ -23,11 +24,85 @@ struct BiomeCompoundData {
     {}
 };
 
+struct ChunkCompoundData {
+public:
+    double amount = 0.0f;
+    std::string name;
+    ChunkCompoundData() {}
+
+    // Move constructor
+    ChunkCompoundData(ChunkCompoundData&& other) :
+        name(std::move(other.name)), amount(other.amount)
+    {}
+
+    // Copy constructor
+    ChunkCompoundData(const ChunkCompoundData& other) :
+        name(other.name), amount(other.amount)
+    {}
+
+    ChunkCompoundData(double amount, std::string name) :
+        amount(amount), name(name)
+    {}
+};
+
+struct ChunkData {
+public:
+    std::string name = "";
+    double density = 1.0f;
+    bool dissolves = true;
+    unsigned int radius = 0;
+    unsigned int mass = 0;
+    unsigned int size = 0;
+    double ventAmount = 3.0f;
+
+    std::vector<std::string> meshes;
+    std::map<size_t, ChunkCompoundData> chunkCompounds;
+
+    // Move constructor
+    ChunkData(ChunkData&& other) :
+        name(std::move(other.name)), density(other.density),
+        dissolves(other.dissolves), radius(other.radius), mass(other.mass),
+        size(other.size), ventAmount(other.ventAmount)
+    {
+        meshes = std::move(other.meshes);
+        chunkCompounds = std::move(other.chunkCompounds);
+    }
+
+    // Copy constructor
+    ChunkData(const ChunkData& other) :
+        name(other.name), density(other.density), dissolves(other.dissolves),
+        radius(other.radius), mass(other.mass), size(other.size),
+        ventAmount(other.ventAmount)
+    {
+        meshes = other.meshes;
```

Comment:
Same thing with the assignment here, should be in the initializer list.

Reply:
did thi snow